### PR TITLE
Add support for golint

### PR DESCRIPTION
**Pull request by rhatdan**
Modify bindings to pass golint.

Make .travis.yml test that pull requests pass golint.

### CODE REVIEW

**Comment by runcom** [comment]
:+1:

**Comment by rhatdan** [comment]
@mrunalp PTAL

File: go-selinux/label/label_selinux.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ var ErrIncompatibleLabel = fmt.Errorf("Bad SELinux option z and Z can not be use
 // the labels.  The labels returned will include a random MCS String, that is
 // guaranteed to be unique.
 func InitLabels(options []string) (string, string, error) {
-	if !selinux.SelinuxEnabled() {
+	if !selinux.GetEnabled() {
```
Comment (mrunalp) [inline go-selinux/label/label_selinux.go]: We should call this just Enabled()
See https://golang.org/doc/effective_go.html#Getters

File: go-selinux/selinux.go [hunk 2]
```diff
@@ -141,8 +145,8 @@ func getSelinuxMountPoint() string {
 	return state.getSELinuxfs()
 }
 
-// SelinuxEnabled returns whether selinux is currently enabled.
-func SelinuxEnabled() bool {
+// GetEnabled returns whether selinux is currently enabled.
+func GetEnabled() bool {
```
Comment (mrunalp) [inline go-selinux/selinux.go]: `GetEnabled -> Enabled`

File: go-selinux/selinux.go [hunk 3]
```diff
@@ -291,7 +314,8 @@ func selinuxEnforcePath() string {
 	return fmt.Sprintf("%s/enforce", selinuxPath)
 }
 
-func SelinuxGetEnforce() int {
+// GetEnforce returns the current SELinux mode Enforcing, Permissive, Disabled
+func GetEnforce() int {
```
Comment (mrunalp) [inline go-selinux/selinux.go]: Maybe call this `EnforceMode()`

File: go-selinux/selinux.go [hunk 4]
```diff
@@ -306,11 +330,20 @@ func SelinuxGetEnforce() int {
...
+SetEnforce sets the current SELinux mode Enforcing, Permissive.
+Disabled is not valid, since this needs to be set at boot time.
+*/
+func SetEnforce(mode int) error {
 	return writeCon(selinuxEnforcePath(), fmt.Sprintf("%d", mode))
 }
 
-func SelinuxGetEnforceMode() int {
+/*
+GetEnforceMode returns the systems default SELinux mode Enforcing,
```
Comment (mrunalp) [inline go-selinux/selinux.go]: Maybe name this `SystemEnforceMode()`

File: go-selinux/selinux.go [hunk 5]
```diff
@@ -393,17 +429,22 @@ func FreeLxcContexts(scon string) {
 
 var roFileLabel string
 
+// GetROFileLabel returns the specified SELinux readonly label
```
Comment (mrunalp) [inline go-selinux/selinux.go]: ROFileLabel

File: go-selinux/selinux.go [hunk 6]
```diff
@@ -393,17 +429,22 @@ func FreeLxcContexts(scon string) {
...
 func GetROFileLabel() (fileLabel string) {
 	return roFileLabel
 }
 
+/*
+GetLxcContexts returns an allocated processLabel and fileLabel to be used for
```
Comment (mrunalp) [inline go-selinux/selinux.go]: `LxcContexts`

File: go-selinux/label/label_selinux.go (hunk 1 above)
Comment (rhatdan) [inline go-selinux/label/label_selinux.go]: The problem is we have a constant called selinux.Enabled already. Which is something a caller might do.

if selinux.GetEnabled() == selinux.Enabled ...

File: go-selinux/selinux.go (hunk 2 above)
Comment (rhatdan) [inline go-selinux/selinux.go]: See Above

File: go-selinux/selinux.go (hunk 3 above)
Comment (rhatdan) [inline go-selinux/selinux.go]: That is fine with me.  Only thing I was thinking was these match up somewhat with the libselinux.so names.  But changing these to 
EnforceMode() 
DefaultEnforceMode()
Would be more descriptive.

File: go-selinux/selinux.go (hunk 4 above)
Comment (rhatdan) [inline go-selinux/selinux.go]: If you like SystemEnforceMode versus DefaultEnforceMode?

File: go-selinux/selinux.go (hunk 6 above)
Comment (rhatdan) [inline go-selinux/selinux.go]: If we are going to change this lets go with ContainerLabels()

File: go-selinux/selinux.go (hunk 3 above)
Comment (mrunalp) [inline go-selinux/selinux.go]: :+1:

File: go-selinux/selinux.go (hunk 4 above)
Comment (mrunalp) [inline go-selinux/selinux.go]: Either is fine with me.

Comment (rhatdan) [inline go-selinux/selinux.go]: DefaultEnforceMode
Fixed

File: go-selinux/selinux.go (hunk 3 above)
Comment (rhatdan) [inline go-selinux/selinux.go]: Fixed

File: go-selinux/selinux.go (hunk 5 above)
Comment (rhatdan) [inline go-selinux/selinux.go]: FIXED

File: go-selinux/selinux.go (hunk 6 above)
Comment (rhatdan) [inline go-selinux/selinux.go]: Fixed

File: go-selinux/selinux.go [hunk 7]
```diff
@@ -384,26 +417,35 @@ func uniqMcs(catRange uint32) string {
...
 }
 
-func FreeLxcContexts(scon string) {
-	if len(scon) != 0 {
-		con := strings.SplitN(scon, ":", 4)
+/*
+FreeLabel will unreserve the MLS/MCS Level field of the specified label.
+Allowing it to be used by another process.
+*/
+func FreeLabel(label string) {
```
Comment (mrunalp) [inline go-selinux/selinux.go]: nit: Should we call this ReleaseLabel?

**Comment by rhatdan** [comment]
@mrunalp changes all calls to ReleaseLabel.  So now we have
ReserveLabel
and 
ReleaseLabel.

**Comment by rhatdan** [comment]
@mrunalp @runcom PTAL

**Comment by mrunalp** [comment]
LGTM

**Comment by runcom** [comment]
LGTM

**Pull request merged**